### PR TITLE
Sign Sashimi.AzureWebApp binaries in cake build

### PR DESCRIPTION
**Pull request by mik-ky**
**Changes**
- Modify cake build script to sign Sashimi.AzureWebApp binaries. Most of the logic is from [Calamari repository.](https://github.com/OctopusDeploy/Calamari)

Verified that the AzureWebApp step is functioning as expected, using the signed binaries from feedz.io:

image

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -190,6 +210,139 @@ Task("CopyToLocalPackages")
...
+{
+    if (BuildSystem.IsLocalBuild && !signFiles) return;
+
+    Information($"Signing binaries in {outputDirectory}");
+
+    // check that any unsigned libraries, that Octopus Deploy authors, get signed to play nice with security scanning tools
+    // refer: https://octopusdeploy.slack.com/archives/C0K9DNQG5/p1551655877004400
+    // decision re: no signing everything: https://octopusdeploy.slack.com/archives/C0K9DNQG5/p1557938890227100
+    var unsignedExecutablesAndLibraries =
+        GetFiles(outputDirectory + "/{Calamari,Sashimi}*.{exe,dll}")
```
Comment (mik-ky) [inline build.cake:233]: Regards to https://github.com/OctopusDeploy/Sashimi.AzureWebApp/pull/27#discussion_r843318023 : the only difference in #27 and #28 should be this line, where it uses a newer file matching, using the current cake tool version.

**Comment by mik-ky** [comment]
@mjhilton This is a newer PR which I've raised over #27. I've tested this locally by upgrading server to use the newer version of Sashimi.AzureWebApp from our feed, and was able to confirm binaries are signed (=image in PR description). 🙇

File: build.ps1 [hunk 2]
```diff
@@ -191,5 +195,5 @@ Write-Host "Installing cake modules using the --bootstrap argument"
 
 # Start Cake
 Write-Host "Running build script..."
-Invoke-Expression "& `"$CAKE_EXE`" `"$Script`" -target=`"$Target`" -configuration=`"$Configuration`" -verbosity=`"$Verbosity`" $UseMono $UseDryRun $UseExperimental $ScriptArgs"
+Invoke-Expression "& `"$CAKE_EXE`" `"$Script`" -target=`"$Target`" -configuration=`"$Configuration`" -verbosity=`"$Verbosity`" -AzureKeyVaultUrl=`"$AzureKeyVaultUrl`" -AzureKeyVaultAppId=`"$AzureKeyVaultAppId`" -AzureKeyVaultAppSecret=`"$AzureKeyVaultAppSecret`" -AzureKeyvaultCertificateName=`"$AzureKeyvaultCertificateName`" $UseMono $UseDryRun $UseExperimental $ScriptArgs"
```
Comment (mjhilton) [inline build.ps1:198]: I notice the capitalisation for `Keyvault` is different here to the other arguments, and to [what's specified in build.cake](https://github.com/OctopusDeploy/Sashimi.AzureWebApp/pull/28/files#diff-c6eace5a70f0c2d920dfa97e988d4cad3d0131dd2b029bb2646c96222e866cf8R25). Does that matter?

**Review by mjhilton — APPROVED** [review]
Looks good, with one query

File: build.ps1 (hunk 2 above)
Comment (mik-ky) [reply build.ps1:198]: Good spot - This is what it is in Calamari at the moment, so I've missed this.

As it's not stated in the cake documentation, I've gone through the cake repository, and it looks like they're doing a case insensitive match on arguments. For example: https://github.com/cake-build/cake/blob/6dd9edec635c99d90ad853d2a4296422f38de70e/src/Cake/CakeArguments.cs

I feel like the capitalisation should be identical regardless - I'm happy to do a clean up PR to address this if you feel like this is worth doing.

**Pull request merged**